src/index.ts

```diff
@@ -632,13 +747,15 @@ export function create (rawOptions: CreateOptions = {}): Register {
         return { name, comment }
       }
     } else {
-      const sys = {
+      const sys: _ts.System & _ts.FormatDiagnosticsHost = {
         ...ts.sys,
         ...diagnosticHost,
         readFile: (fileName: string) => {
           const cacheContents = fileContents.get(fileName)
           if (cacheContents !== undefined) return cacheContents
-          return cachedReadFile(fileName)
+          const contents = cachedReadFile(fileName)
+          if (contents) fileContents.set(fileName, contents)
+          return contents
```

Comment:
We use `fileContents` elsewhere to know if the compiler thinks a file has "changed."  If the compiler observes a file here, then we need to remember so that our other logic correctly realizes the file has not changed.